src/chunk.c

```diff
@@ -2603,7 +2608,7 @@ chunk_tuple_delete(TupleInfo *ti, DropBehavior behavior, bool preserve_chunk_cat
 
 	if (!preserve_chunk_catalog_row)
 	{
-		ts_catalog_delete(ti->scanrel, ti->tuple);
+		ts_catalog_delete_tid(ti->scanrel, ts_scanner_get_tuple_tid(ti));
```

Comment:
I see some places where ts_catalog_delete has been replaced by ts_catalog_delete_tid. What is the reasoning behind the change? Aren't they equivalent in this case?

Reply:
The table API wraps tuples in tuple slots, so we don't have direct access to the tuple unless it is potentially re-materialized, which comes at a cast. FWIW, `ts_catalog_delete` takes a tuple, but internally just calls `ts_catalog_delete_tid`.